### ISSUE TITLE
self hosting the UI?

### ISSUE THREAD

**Issue by InteNs** [chdsbd/kodiak#684]
Hey there,

We have been using a self-hosted version of the kodiak bot with great success.

Is it possible to self-host the UI as well? It's not documented anywhere so i'm not entirely sure where to start.

The main reason we want this is so that we can see the queue.

Thanks!

**Comment by chdsbd** [comment]
Hi @InteNs,

While it is possible to self-host the UI, it's not well documented. There's some Ansible scripts in [`infrastructure/`](https://github.com/chdsbd/kodiak/tree/master/infrastructure) that might be useful.

There's a few things that need to be configured:

* Load balancer like HAproxy to route traffic to the Web UI or API
* Postgres instance for the Web API
* OAuth connection for GitHub login
* Web api, pull ingestion service, and web ui

Basically it's a pain to setup and manage

**Comment by houms-sony** [comment]
@chdsbd just wanted to see if I could ask some follow-ups regarding this setup. We are in the same boat, but due to compliance restrictions we would like to host the ui ourselves. Not sure if I should open a separate issue or just add to this one?

from what we have gathered so far,  we need
- lb to route traffic to web ui/api
- oauth app in github
- postgres instance for the web api 
- web ui  container runs nginx for static assets
- web api container runs the gunicorn service for web api
- separate web api container runs the ingestor service (persistent)
- separate web api containers for two cron jobs (aggregating user activity and events into chartable data)

We have setup the lb, oauth app in github,  postgres, web ui and api(gunicorn), but not the ingestor or the two cron jobs. 
we are exposing web ui on app.domain.com/
we are exposting web api on app.domain.com/v1
We are having the following issues; 
When we visit our app.domain.com we get redirected to accounts page with "failed to load accounts data" 
If we go to app.domain.com/login we get the "Login with Github" button", but when we click it we get a mismatch on callback url  (we tried using app.domain.com/v1/login)

Is it safe to assume the web ui and api served on the same public domain name is fine?
what routes need to be exposed publicly for web api and ui?
in re github app for auth, what is the authorized callback url needed?
the readme for web-ui says copy example.env but there is none in the web_ui dir., is this correct, does the web-ui container not require a .env?



Please let me know if you would be willing to provide some additional info/pointers? Thanks for your help.

**Comment by chdsbd** [comment]
@houms-sony Feel free to reach out to me at chris@kodiakhq.com for some help
